pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -1,203 +1,688 @@
+    static final int TAG_TYPE_BITS = 3;
+
+    /** Wire format code for var int */
+    public static final int TYPE_VARINT = 0;
+    /** Wire format code for fixed 64bit number */
+    public static final int TYPE_FIXED64 = 1;
+    /** Wire format code for length delimited, all the complex types */
+    public static final int TYPE_LENGTH_DELIMITED = 2;
+    /** Wire format code for fixed 32bit number */
+    public static final int TYPE_FIXED32 = 5;
+
+    /**
+     * Makes a tag value given a field number and wire type.
+     *
+     * @param wireType the wire type part of tag
+     * @param fieldNumber the field number part of tag
+     * @return packed encoded tag
+     */
+    public static int getTag(final int wireType, final int fieldNumber) {
+        return (fieldNumber << TAG_TYPE_BITS) | wireType;
+    }
+
+    /**
+     * Make sure first character of a string is upper case
+     *
+     * @param name string input who's first character can be upper or lower case
+     * @return name with first character converted to upper case
+     */
+    public static String capitalizeFirstLetter(String name) {
+        if (name.length() > 0) {
+            if (name.chars().allMatch(Character::isUpperCase)) {
+                return Character.toUpperCase(name.charAt(0)) + name.substring(1).toLowerCase();
+            } else {
+                return Character.toUpperCase(name.charAt(0)) + name.substring(1);
+            }
+        }
+        return name;
+    }
+
+    /**
+     * Convert names like "hello_world" to "HelloWorld" or "helloWorld" depending on firstUpper.
+     * Also handles special case like "HELLO_WORLD" to same output as "hello_world", while
+     * "HelloWorld_Two" still becomes "helloWorldTwo".
+     *
+     * @param name input name in snake case
+     * @param firstUpper if true then first char is upper case otherwise it is lower
+     * @return out name in camel case
+     */
+    public static String snakeToCamel(String name, boolean firstUpper) {
+        final String out =
+                Arrays.stream(name.split("_"))
+                        .map(Common::capitalizeFirstLetter)
+                        .collect(Collectors.joining(""));
+        return (firstUpper
+                        ? Character.toUpperCase(out.charAt(0))
+                        : Character.toLowerCase(out.charAt(0)))
+                + out.substring(1);
+    }
+
+    /**
+     * Convert a camel case name to upper case snake case
+     *
+     * @param name the input name in camel case
+     * @return output name in upper snake case
+     */
+    public static String camelToUpperSnake(String name) {
+        // check if already camel upper
+        if (name.chars()
+                .allMatch(c -> Character.isUpperCase(c) || Character.isDigit(c) || c == '_'))
+            return name;
+        // check if already has underscores, then just capitalize
+        if (name.chars().anyMatch(c -> c == '_')) return name.toUpperCase();
```

Comment:
`name.contains()` ?

Reply:
Again... Just spotless formatting modification. But yes, I think contains would be faster probably, instead of having a routine we call for each character.